src/ng/compile.js

```diff
@@ -779,6 +779,14 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
     return bindings;
   }
 
+  function assertFirstLetterIsLowerCased(name) {
+    var letter = name.charAt(0);
+    if (letter !== letter.toLowerCase()) {
+      throw ngMinErr('badname', "Directive '{0}' is not a valid name. Make sure the first letter is lowercase");
```

Comment:
Rewrite the error to "The directive name '{0}' is invalid. The first letter of a directive must be a low case letter"